axelor-base/src/main/resources/views/Product.xml

```diff
@@ -101,7 +101,7 @@
                         onChange="action-product-validate-sale-supply-select"
                         hideIf="isShippingCostsProduct"/>
 					<panel name="units" title="Units" colSpan="6">
-						<field name="unit" canEdit="false" colSpan="12" form-view="unit-form" grid-view="unit-grid" readonlyIf="productTypeSelect = 'storable'"/>
+						<field name="unit" canEdit="false" colSpan="12" form-view="unit-form" grid-view="unit-grid" readonlyIf="productTypeSelect == 'storable'"/>
```

Comment:
Remove this wrong condition. We should be able to define the stock unit for storable product. However, once we have filled a stock location, we shouln't be able to change the unit.